modules/light-clients/xx-mock/types/client_state.go

```diff
@@ -47,299 +58,155 @@ func (cs ClientState) Status(_ sdk.Context, _ sdk.KVStore, _ codec.BinaryCodec)
-	prefix exported.Prefix,
-	proof []byte,
-	consensusState exported.ConsensusState,
-) error {
-	// NOTE In cosmos/ibc-go, it cannot give a consensus state of an external prover(e.g. mock-client) to the client, so we skip this verification for now.
 	return nil
 }
 
-// VerifyConnectionState verifies a proof of the connection state of the
-// specified connection end stored on the target machine.
-func (cs ClientState) VerifyConnectionState(
-	store sdk.KVStore,
+// VerifyNonMembership is a generic proof verification method which verifies the absence of a given CommitmentPath at a specified height.
+// The caller is expected to construct the full CommitmentPath from a CommitmentPrefix and a standardized path (as defined in ICS 24).
+func (cs ClientState) VerifyNonMembership(
+	ctx sdk.Context,
+	clientStore sdk.KVStore,
 	cdc codec.BinaryCodec,
 	height exported.Height,
-	prefix exported.Prefix,
+	delayTimePeriod uint64,
+	delayBlockPeriod uint64,
 	proof []byte,
-	connectionID string,
-	connectionEnd exported.ConnectionI,
+	path exported.Path,
 ) error {
-	_, err := produceVerificationArgs(store, cdc, cs, height, prefix, proof)
-	if err != nil {
-		return err
-	}
-
-	connection, ok := connectionEnd.(connectiontypes.ConnectionEnd)
-	if !ok {
+	if cs.GetLatestHeight().LT(height) {
 		return sdkerrors.Wrapf(
-			connectiontypes.ErrInvalidConnection,
-			"expected type %T, got %T", connectiontypes.ConnectionEnd{}, connectionEnd,
+			sdkerrors.ErrInvalidHeight,
+			"client state height < proof height (%d < %d), please ensure the client has been updated", cs.GetLatestHeight(), height,
 		)
 	}
 
-	bz, err := cdc.Marshal(&connection)
-	if err != nil {
-		return err
-	}
-
-	h := sha256.Sum256(bz)
-	if !bytes.Equal(proof, h[:]) {
-		return fmt.Errorf("expected the proof '%X', actually got '%X'", proof, h)
-	}
-	return nil
-}
-
-// VerifyChannelState verifies a proof of the channel state of the specified
-// channel end, under the specified port, stored on the target machine.
-func (cs ClientState) VerifyChannelState(
-	store sdk.KVStore,
-	cdc codec.BinaryCodec,
-	height exported.Height,
-	prefix exported.Prefix,
-	proof []byte,
-	portID,
-	channelID string,
-	channelEnd exported.ChannelI,
-) error {
-	_, err := produceVerificationArgs(store, cdc, cs, height, prefix, proof)
-	if err != nil {
+	if err := verifyDelayPeriodPassed(ctx, clientStore, height, delayTimePeriod, delayBlockPeriod); err != nil {
 		return err
 	}
 
-	channel, ok := channelEnd.(channeltypes.Channel)
-	if !ok {
-		return sdkerrors.Wrapf(
-			channeltypes.ErrInvalidChannel,
-			"expected channel type %T, got %T", channeltypes.Channel{}, channelEnd)
+	if _, found := getConsensusState(clientStore, cdc, height); !found {
+		return sdkerrors.Wrap(clienttypes.ErrConsensusStateNotFound, "please ensure the proof was constructed against a height that exists on the client")
 	}
 
-	bz, err := cdc.Marshal(&channel)
-	if err != nil {
-		return err
+	if len(proof) != 0 {
```

Comment:
NOTE: I understood that you followed solidity mock-client impl in https://github.com/hyperledger-labs/yui-ibc-solidity/blob/main/contracts/clients/MockClient.sol#L166

Reply:
Yes. Was it the right choice?

Reply:
Yes, you are right :)